providers/azure.go

```diff
@@ -151,6 +157,60 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (s
 		RefreshToken: jsonResponse.RefreshToken,
 	}
 	return
+
+}
+
+// RefreshSessionIfNeeded checks if the session has expired and uses the
+// RefreshToken to fetch a new ID token if required
+func (p *AzureProvider) RefreshSessionIfNeeded(ctx context.Context, s *sessions.SessionState) (bool, error) {
+	if s == nil || s.ExpiresOn.Add(-time.Minute*10).After(time.Now()) || s.RefreshToken == "" {
+		return false, nil
+	}
+
+	origExpiration := s.ExpiresOn
+
+	err := p.redeemRefreshToken(ctx, s)
+
+	if err != nil {
+		return false, fmt.Errorf("unable to redeem refresh token: %v", err)
+	}
+
+	fmt.Printf("refreshed id token %s (expired on %s)\n", s, origExpiration)
+	return true, nil
+}
+
+func (p *AzureProvider) redeemRefreshToken(ctx context.Context, s *sessions.SessionState) (err error) {
```

Comment:
NIT: We're trying to move away from named return parameters in new code, but I see this convention is used all over this provider -- so we can stick with it if you prefer.

Reply:
I'd prefer if we don't add named returns even if they are common across the provider, can we update this one?